File: output/ChatGPT/avatar/Python/C++/codeforces_96_B.cpp
#include <iostream>
#include <vector>
using namespace std;

long long limit = 10000000000;
vector<long long> l;

void gen(long long number, int four, int seven) {
    if (number > limit) {
        return;
    }
    if (number > 0 && four == seven) {
        l.push_back(number);
    }
    gen(number * 10 + 4, four + 1, seven);
    gen(number * 10 + 7, four, seven + 1);
}

int main() {
    gen(0, 0, 0);
    sort(l.begin(), l.end());
    long long n;
    cin >> n;
    long long ans = 0;
    for (long long val : l) {
        if (val >= n) {
            ans = val;
            break;
        }
    }
    cout << ans << endl;
    return 0;
}

File: output/CodeLlama/avatar/Python/Java/codeforces_96_B.java
import java.util.Scanner;
import java.util.ArrayList;
import java.util.Collections;
import java.util.Comparator;

public class codeforces_96_B {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        ArrayList<Integer> l = new ArrayList<>();
        int limit = 10000000000;
        gen(0, 0, 0, l);
        Collections.sort(l, new Comparator<Integer>() {
            @Override
            public int compare(Integer o1, Integer o2) {
                return o1 - o2;
            }
        });
        int n = sc.nextInt();
        int ans = 0;
        for (int val : l) {
            if (val >= n) {
                ans = val;
                break;
            }
        }
        System.out.println(ans);
    }

    public static void gen(int number, int four, int seven, ArrayList<Integer> l) {
        if (number > limit) {
            return;
        }
        if (number > 0 && four == seven) {
            l.add(number);
        }
        gen(number * 10 + 4, four + 1, seven, l);
        gen(number * 10 + 7, four, seven + 1, l);
    }
}